sql/fxa_derived/fxa_amplitude_export_v1/query.sql

```diff
@@ -0,0 +1,100 @@
+WITH
+  base_events AS(
+  SELECT
+    *
+  FROM
+    `moz-fx-fxa-prod-0712.fxa_prod_logs.docker_fxa_auth_20*`
+  WHERE
+    _TABLE_SUFFIX = FORMAT_DATE('%g%m%d', @submission_date)
+    AND jsonPayload.fields.event_type IN ( 'fxa_activity - cert_signed',
+      'fxa_activity - access_token_checked',
+      'fxa_activity - access_token_created' )
+    AND jsonPayload.fields.user_id IS NOT NULL ),
+  grouped_by_user AS (
+  SELECT
+    -- to prevent weirdness from timestamp field, use provided
+    -- submission date parameter as timestamp
+    CAST(@submission_date AS DATETIME) AS submission_timestamp,
+    -- TODO: JS udf with HMAC
+    TO_HEX(SHA256(jsonPayload.fields.user_id)) AS user_id,
+    MIN(CONCAT(insertId, '-user')) AS insert_id,
+    CAST(@submission_date AS DATETIME) AS timestamp,
+    -- Amplitude properties, scalars
+    `moz-fx-data-shared-prod`.udf.mode_last(ARRAY_AGG(jsonPayload.fields.region)) AS region,
+    `moz-fx-data-shared-prod`.udf.mode_last(ARRAY_AGG(jsonPayload.fields.country)) AS country,
+    `moz-fx-data-shared-prod`.udf.mode_last(ARRAY_AGG(jsonPayload.fields.LANGUAGE)) AS LANGUAGE,
+    -- Event properties, arrays
+    ARRAY_AGG(DISTINCT JSON_EXTRACT_SCALAR(jsonPayload.fields.event_properties,
+        "$.service") IGNORE NULLS) AS services,
+    ARRAY_AGG(DISTINCT JSON_EXTRACT_SCALAR(jsonPayload.fields.event_properties,
+        "$.oauth_client_id") IGNORE NULLS) AS oauth_client_ids,
+    -- User properties, arrays
+    ARRAY_AGG(DISTINCT JSON_EXTRACT_SCALAR(jsonPayload.fields.user_properties,
+        "$['$append'].fxa_services_used") IGNORE NULLS) AS fxa_services_used,
+    ARRAY_AGG(DISTINCT jsonPayload.fields.os_name IGNORE NULLS) AS os_used_day,
```

Comment:
Leif, maybe we could get the FxA team to pull e.g. `os_used_week` and `os_used_month`

Reply:
I think Alex has talked about wanting that in the past. I'm not sure we would block this on getting it though? I can ask him.

Reply:
Nope, but we could always add it later. Just a thought.

Reply:
The os_used_week and os_used_month additions here greatly increase the complexity of the logic, and it's not clear to me what these mean. Can you add some code comments to explain the intention of maintaining these arrays?

Reply:
From what I can gather so far about these arrays of OS names, I think this would be better served not as nested arrays but instead using bit patterns. This would allow more reuse of existing UDFs and reduce the need for new abstractions. We would have a single field like:

    os_usage ARRAY<STRUCT<os STRING, days_used_bits INT64>>

Then, you would be able to derive the `os_used_week` and `os_used_month` arrays from that set of bit patterns.